src/library_glfw.js

```diff
@@ -1354,15 +1418,43 @@ var LibraryGLFW = {
 
   glfwCreateWindowSurface: function(instance, winid, allocator, surface) { throw "glfwCreateWindowSurface is not implemented."; },
 
-  glfwSetJoystickCallback: function(cbfun) { throw "glfwSetJoystickCallback is not implemented."; },
+  glfwJoystickPresent: function(joy) {
+    return GLFW.active.joys[joy] !== undefined;
+  },
+
+  glfwGetJoystickAxes: function(joy, count) {
+    GLFW.refreshJoystick(joy);
 
-  glfwJoystickPresent: function(joy) { throw "glfwJoystickPresent is not implemented."; },
+    var state = GLFW.active.joys[joy];
+    if (!state || !state.axes) {
+      setValue(count, 0, 'i32');
+      return;
+    }
 
-  glfwGetJoystickAxes: function(joy, count) { throw "glfwGetJoystickAxes is not implemented."; },
+    setValue(count, state.axesCount, 'i32');
+    return state.axes;
+  },
 
-  glfwGetJoystickButtons: function(joy, count) { throw "glfwGetJoystickButtons is not implemented."; },
+  glfwGetJoystickButtons: function(joy, count) {
+    GLFW.refreshJoystick(joy);
 
-  glfwGetJoystickName: function(joy) { throw "glfwGetJoystickName is not implemented."; },
+    var state = GLFW.active.joys[joy];
+    if (!state || !state.buttons) {
+      setValue(count, 0, 'i32');
+      return;
+    }
+
+    setValue(count, state.buttonsCount, 'i32');
+    return state.buttons;
+  },
+
+  glfwGetJoystickName: function(joy) {
+    return GLFW.active.joys[joy].id;
```

Comment:
If joystick is disconnected right before this function is called, `GLFW.active.joys[joy]` could be deleted above and be undefined, making this function throw an exception on `undefined.id` access; so check for `if (GLFW.active.joys[joy])` here before proceeding?